cvat-canvas3d/src/typescript/canvas3dView.ts

```diff
@@ -73,14 +201,90 @@ export class Canvas3dViewImpl implements Canvas3dView, Listener {
+        Canvas3dViewImpl.resizeRendererToDisplaySize(this.views.top);
+        Canvas3dViewImpl.resizeRendererToDisplaySize(this.views.side);
+        Canvas3dViewImpl.resizeRendererToDisplaySize(this.views.front);
+        this.views.perspective.renderer.render(this.views.perspective.scene, this.views.perspective.camera);
+        this.views.top.renderer.render(this.views.top.scene, this.views.top.camera);
+        this.views.side.renderer.render(this.views.side.scene, this.views.side.camera);
+        this.views.front.renderer.render(this.views.front.scene, this.views.front.camera);
+    }
+
+    public keyControls(key: any): void {
+        if (key.altKey === true) {
+            switch (key.code) {
+                case CAMERA_ACTION.ZOOM_IN:
+                    this.views.perspective.controls.dolly(5, true);
+                    break;
+                case CAMERA_ACTION.ZOOM_OUT:
+                    this.views.perspective.controls.dolly(-5, true);
+                    break;
+                case CAMERA_ACTION.MOVE_LEFT:
+                    this.views.perspective.controls.truck(-0.01 * this.speed, 0, true);
+                    break;
+                case CAMERA_ACTION.MOVE_RIGHT:
+                    this.views.perspective.controls.truck(0.01 * this.speed, 0, true);
+                    break;
+                case CAMERA_ACTION.MOVE_DOWN:
+                    this.views.perspective.controls.truck(0, -0.01 * this.speed, true);
+                    break;
+                case CAMERA_ACTION.MOVE_UP:
+                    this.views.perspective.controls.truck(0, 0.01 * this.speed, true);
+                    break;
+                default:
+                    break;
+            }
+        } else {
+            switch (key.code) {
+                case CAMERA_ACTION.ROTATE_RIGHT:
+                    this.views.perspective.controls.rotate(0.1 * THREE.MathUtils.DEG2RAD * this.speed, 0, true);
+                    break;
+                case CAMERA_ACTION.ROTATE_LEFT:
+                    this.views.perspective.controls.rotate(-0.1 * THREE.MathUtils.DEG2RAD * this.speed, 0, true);
+                    break;
+                case CAMERA_ACTION.TILT_UP:
+                    this.views.perspective.controls.rotate(0, -0.05 * THREE.MathUtils.DEG2RAD * this.speed, true);
+                    break;
+                case CAMERA_ACTION.TILT_DOWN:
+                    this.views.perspective.controls.rotate(0, 0.05 * THREE.MathUtils.DEG2RAD * this.speed, true);
+                    break;
+                default:
+                    break;
+            }
+        }
     }
 
-    public html(): any {
-        return this.renderer.domElement;
+    public html(): ViewsDOM {
+        return {
+            perspective: this.views.perspective.renderer.domElement,
+            top: this.views.top.renderer.domElement,
+            side: this.views.side.renderer.domElement,
+            front: this.views.front.renderer.domElement,
+        };
     }
 }
```

Comment:
In general refactor please with consts for "magic numbers", destructions and loops where it's possible

Reply:
Resolved